src/libs/IOUUtils.js

```diff
@@ -65,7 +66,77 @@ function updateIOUOwnerAndTotal(iouReport, actorEmail, amount, currency, type =
+ * Used below so that we can decide if an IOU report is pending currency conversion.
+ *
+ * @param {Array} reportActions
+ * @param {Object} iouReport
+ * @param {String} type - iouReportAction type. Can be oneOf(create, decline, cancel, pay, split)
+ * @param {String} pendingAction
+ * @param {Boolean} filterRequestsInDifferentCurrency
+ *
+ * @returns {Array}
+ */
+function getIOUReportActions(reportActions, iouReport, type = '', pendingAction = '', filterRequestsInDifferentCurrency = false) {
+    return _.chain(reportActions)
+        .filter(action => action.originalMessage
+            && action.actionName === CONST.REPORT.ACTIONS.TYPE.IOU
+            && action.originalMessage.IOUReportID.toString() === iouReport.reportID.toString())
+        .filter(action => (!_.isEmpty(type) ? action.originalMessage.type === type : true))
+        .filter(action => (!_.isEmpty(pendingAction) ? action.pendingAction === pendingAction : true))
+        .filter(action => (filterRequestsInDifferentCurrency ? action.originalMessage.currency !== iouReport.currency : true))
+        .value();
+}
+
+/**
+ * Returns whether or not an IOU report contains money requests in a different currency
+ * that are either created or cancelled offline, and thus haven't been converted to the report's currency yet
+ *
+ * @param {Array} reportActions
+ * @param {Object} iouReport
+ *
+ * @returns {Boolean}
+ */
+function isIOUReportPendingCurrencyConversion(reportActions, iouReport) {
+    // Pending money requests that are in a different currency
+    const pendingRequestsInDifferentCurrency = _.chain(getIOUReportActions(
+        reportActions,
+        iouReport,
+        CONST.IOU.REPORT_ACTION_TYPE.CREATE,
+        CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+        true,
+    )).map(action => action.originalMessage.IOUTransactionID)
+        .sort()
+        .value();
+
+    // Pending cancelled money requests that are in a different currency
+    const pendingCancelledRequestsInDifferentCurrency = _.chain(getIOUReportActions(
+        reportActions,
+        iouReport,
+        CONST.IOU.REPORT_ACTION_TYPE.CANCEL,
+        CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+        true,
+    )).map(action => action.originalMessage.IOUTransactionID)
+        .sort()
+        .value();
+
+    const hasPendingRequests = Boolean(pendingRequestsInDifferentCurrency.length || pendingCancelledRequestsInDifferentCurrency.length);
+
+    // If we have pending money requests made offline, check if all of them have been cancelled offline
+    // In order to do that, we can grab transactionIDs of all the created and cancelled money requests and check if they're identical
+    if (hasPendingRequests && _.isEqual(pendingRequestsInDifferentCurrency, pendingCancelledRequestsInDifferentCurrency)) {
+        return false;
+    }
+
+    // We don't have requests made offline that have all been cancelled
+    // simply return if we have any pending created or cancelled requests
```

Comment:
NAB: But for some reason this comment is really hard for me to understand. This suggestion is not much better, so open to anything else, simpler which will capture the essence of this.
```suggestion
    // Not all requests made offline had been cancelled,
    // simply return if we have any pending created or cancelled requests
```